src/components/SingleEvent/Admin/PartyForm.js

```diff
@@ -415,23 +416,53 @@ class PartyForm extends Component {
             mutation={mutation}
             resultKey="id"
             variables={variables}
-            onCompleted={
-              onCompleted
-                ? ({ id }) =>
-                    onCompleted(
-                      { id },
-                      deposit,
-                      limitOfParticipants,
-                      coolingPeriod
+          >
+            {mutate =>
+              type === 'create' ? (
+                <ChainMutation mutation={CREATE_PARTY} resultKey="create">
+                  {(createParty, result) => {
+                    const address = result.data
+                      ? extractNewPartyAddressFromTx(result.data)
+                      : null
+
+                    return (
+                      <>
+                        <ChainMutationButton
+                          analyticsId="Deploy Event Contract"
+                          result={result}
+                          onClick={() => {
+                            mutate().then(({ data: { id } }) => {
+                              createParty({
+                                variables: {
+                                  id,
+                                  deposit,
+                                  limitOfParticipants,
+                                  coolingPeriod
+                                }
+                              })
+                            })
+                          }}
+                          preContent={getButtonText(type)}
+                          postContent="Deployed!"
+                        />
+                        {address ? (
+                          <p>
+                            Event at {address}!{' '}
+                            <Link to={`/event/${address}`}>
```

Comment:
ah ok, you created a message to show the new address but I cannot see it because responding on transaction confirmation is broken.?

Reply:
yes